### PR TITLE
Translate to Spanish

### PR DESCRIPTION
**Pull request by DavidMedinaO**
## Translate to Spanish:

Servant pattern
Sharding pattern
Spatial-partition pattern
Special-case pattern
specification pattern
State pattern
Strategy pattern
Subclass-case pattern

### CODE REVIEW

File: localization/es/spatial-partition/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Esto incluirá un montón de controles innecesarios entre jugadores que están demasiado alejados como para tener alguna
+influencia entre ellos. Los bucles anidados dan a esta operación una complejidad O(n^2), que tiene que ser
+cada fotograma, ya que muchos de los objetos del campo pueden moverse en cada fotograma. La idea
+detrás del patrón de diseño de Partición Espacial es permitir una rápida localización de los objetos utilizando una estructura de datos que está organizada por sus posiciones.
+que está organizada por sus posiciones, de modo que cuando se realiza una operación como la anterior,
+no es necesario cotejar la posición de cada objeto con la de todos los demás. La estructura de datos
+puede utilizarse para almacenar objetos en movimiento y estáticos, aunque para seguir la pista de los objetos en movimiento,
+sus posiciones tendrán que restablecerse cada vez que se muevan. Esto significaría tener que crear una nueva
+instancia de la estructura de datos cada vez que un objeto se mueva, lo que consumiría memoria adicional. La dirección
+estructuras de datos comunes utilizadas para este patrón de diseño son:
```
Comment (luishincapiebf) [inline localization/es/spatial-partition/README.md:47]: Hey David pls review this translation, some lines seem duplicated

File: localization/es/specification/README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+## También conocido como
+
+Filter, Criteria
+
+## Propósito
+
+Specification pattern separates the statement of how to match a candidate, from the candidate object 
+that it is matched against. As well as its usefulness in selection, it is also valuable for 
+validation and for building to order.
```
Comment (luishincapiebf) [inline localization/es/specification/README.md:17]: missing translation

Comment (luismateoh) [inline localization/es/specification/README.md:17]: Missing translation

File: localization/es/state/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+## Diagrama de clases
+
+![alt text](./etc/state_urm.png "State")
+
+## Aplicabilidad
+
+Use the State pattern in either of the following cases
+
+* An object's behavior depends on its state, and it must change its behavior at run-time depending on that state
+* Operations have large, multipart conditional statements that depend on the object's state. This state is usually represented by one or more enumerated constants. Often, several operations will contain this same conditional structure. The State pattern puts each branch of the conditional in a separate class. This lets you treat the object's state as an object in its own right that can vary independently from other objects.
```
Comment (luismateoh) [inline localization/es/state/README.md:154]: Missing translation

File: localization/es/strategy/README.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+---
+
+## También conocido como
+
+Policy
+
+## Propósito
+
+Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets 
+the algorithm vary independently of the clients that use it.
```
Comment (luismateoh) [inline localization/es/strategy/README.md:16]: Misisng translation

File: localization/es/specification/README.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  @Override
+  public boolean test(T t) {
+    return leafComponents.stream().allMatch(comp -> (comp.test(t)));
+  }
+}
+```
+
+Todo lo que queda por hacer ahora es crear selectores de hoja (ya sean de código duro o parametrizados) que
+sean lo más genéricos posible, y podremos instanciar la clase ``AbstractSelector`` combinando
+combinando cualquier cantidad de selectores, como se ejemplificó anteriormente. Debemos tener cuidado, sin embargo, ya que es fácil
```
Comment (luismateoh) [inline localization/es/specification/README.md:180]: Reapeated word 'combinando'

File: localization/es/spatial-partition/README.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Como se explica en el libro  [Game Programming Patterns](http://gameprogrammingpatterns.com/spatial-partition.html) 
+por Bob Nystrom, spatial partition pattern ayuda a localizar eficazmente los objetos almacenándolos en una
+estructura de datos organizada por sus posiciones.
+
+## Explicación
+
+Digamos que usted está construyendo un juego de guerra con cientos, o incluso miles de jugadores, que se enfrentan en el campo de batalla.
+en el campo de batalla. La posición de cada jugador se actualiza en cada fotograma. La forma más sencilla de manejar
+todas las interacciones que tienen lugar en el campo es comprobar la posición de cada jugador con la de todos los demás.
+posición de cada jugador:
```
[the remark this answers is not in the document]
Comment (DavidMedinaO) [reply localization/es/spatial-partition/README.md:21]: Got it, thank you, I'm going to do it.

File: localization/es/specification/README.md (hunk 2 above)
Comment (DavidMedinaO) [reply localization/es/specification/README.md:17]: Got it

File: localization/es/state/README.md (hunk 3 above)
Comment (DavidMedinaO) [reply localization/es/state/README.md:154]: Got it, I'm going to do it.

File: localization/es/strategy/README.md (hunk 4 above)
Comment (DavidMedinaO) [reply localization/es/strategy/README.md:16]: Got it

File: localization/es/specification/README.md (hunk 5 above)
Comment (DavidMedinaO) [reply localization/es/specification/README.md:180]: Got it,  I'll fix it.

Comment (DavidMedinaO) [reply localization/es/specification/README.md]: Done

File: localization/es/strategy/README.md (hunk 4 above)
Comment (DavidMedinaO) [reply localization/es/strategy/README.md]: Done

File: localization/es/state/README.md (hunk 3 above)
Comment (DavidMedinaO) [reply localization/es/state/README.md]: Done

File: localization/es/specification/README.md (hunk 2 above)
Comment (DavidMedinaO) [reply localization/es/specification/README.md]: Done

File: localization/es/spatial-partition/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (DavidMedinaO) [reply localization/es/spatial-partition/README.md]: Done

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @DavidMedinaO for translation

**Pull request merged**